Redirect all files in /examples folder to kubernetes/examples repo

sebgoa opened the pull request:
**What this PR does / why we need it**:

Examples are being moved to their own repository: https://github.com/kubernetes/examples

We need to remove them from the main repo , but first we need to keep a redirect.

This is a *big* organizational change, but nothing technical (aside from e2e tests)

**Which issue this PR fixes** 

fixes part of #24343 

**Special notes for your reviewer**:

WIP, I still need to figure out what to do with the BUILD script and tests, plus take care of the e2e tests that use some of these examples.

**release notes**
```release-note
Redirect all examples README to the the kubernetes/examples repo
```

sebgoa commented:
cc/ @kubernetes/examples big change coming here

ahmetb commented:
Sent https://github.com/kubernetes/examples/pull/5, maybe we can merge that one first and then update the paths that are not in `staging/` in a single PR.

ahmetb commented:
This is gonna be interesting... Have we ever moved out Go code and/or YAML files? Replacing all files with the same test is causing Go builds to break (also would break any tutorials referencing to their YAML files or `kubectl create -f https://...`.

What if we did this: If there's a README.md in the directory, replace it with the migration notice. If there's no README.md in a directory, add a README.md with the migration notice, leave all other files untouched?

sebgoa commented:
yes, I wanted to get this out to get some feedback and see what all it would break.

thockin commented on examples/meteor/README.md:
```diff
@@ -1,215 +1 @@
-<!-- BEGIN MUNGE: GENERATED_ANALYTICS -->
-[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/examples/meteor/README.md?pixel)]()
-<!-- END MUNGE: GENERATED_ANALYTICS -->
+This file has moved to [https://github.com/kubernetes/examples/blob/master/staging/teor/README.md](https://github.com/kubernetes/examples/blob/master/staging/teor/README.md)
```
Some of these links are broken - I just picked this one at random.

sebgoa committed:
add redirect notice in all readme files

sebgoa commented:
@ahmetb ok I rebased this and just added a notice in the README.

I think it is a good step to start directing folks to the other repo. Once this gets merged we can clean up further.

ahmetb commented:
/lgtm

I like how you only modified only the README.md files. Otherwise it would break many docs/tutorials in k8s.io and other places. Perhaps delete the `guidelines.md` as well? Also what do you think of using `git.k8s.io/examples` format instead of github URLs? I think this feature is rather new.

sebgoa commented:
/assign @thockin

ahmetb commented:
@sebgoa is this still WIP as the title indicates? If not, we should probably request for some help to move this forward.

From what I can tell there  are already conflicting files (meaning = `examples/` patches are still being merged to this repo instead of https://github.com/kubernetes/examples). We should at least try blocking (non-autogenerated) PRs to `examples/`.

```
examples/cockroachdb/README.md
examples/persistent-volume-provisioning/README.md
```

sebgoa commented:
PR rebased. We need to get this merged so that people will stop merging examples and move to https://github.com/kubernetes/examples

I will try to join the k8s community hangout to talk about it briefly.

then we will need to:

* deal with the current issues filed for examples
* clean up the build
* remove references to these examples in the API docs
* figure out the docs import of these examples

/assign @brendanburns who is an approver

and let's cc @kubernetes/sig-contributor-experience-bugs since the goal is to break up the repos and gain contribution momentum on the examples.

ahmetb commented:
/lgtm

sebgoa commented:
/retest

kargakis commented:
fyi @kubernetes/sig-testing-misc since this was a discussion topic in the last call

grodrigues3 commented:
@brendanburns PTAL.

sebgoa commented:
@thockin Hi Tim, sorry to ping you directly, but I am out of ideas on how to get attention to this.

thanks

timothysc commented on examples/elasticsearch/README.md:
```diff
@@ -1,163 +1 @@
-<!-- BEGIN MUNGE: GENERATED_ANALYTICS -->
-[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/examples/elasticsearch/README.md?pixel)]()
-<!-- END MUNGE: GENERATED_ANALYTICS -->
+This file has moved to [https://github.com/kubernetes/examples/blob/master/staging/elasticsearch/README.md](https://github.com/kubernetes/examples/blob/master/staging/elasticsearch/README.md)
```
I don't understand why there isn't just a single readme at the root, everything else seems redundant.

sebgoa replied:
@timothysc we went with the same concept than when the docs/devel were moved. We leave breadcrumbs in case people linked directly to the repo. that way they know where to go:

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md

timothysc replied:
Well I'm ok with it for now just to get it out of the repo.

sebgoa replied:
good I have been trying to get this merged sine May 24.

timothysc commented:
@sebgoa could you resolve the conflicts and I will green light this 2day. 

I want these things out ASAP.

sebgoa commented:
rebased
